src/client/Microsoft.Identity.Client/OAuth2/TokenClient.cs

```diff
@@ -169,31 +167,52 @@ private void AddBodyParamsAndHeaders(IDictionary<string, string> additionalBodyP
                     _serviceBundle.HttpTelemetryManager.GetLastRequestHeader());
             }
 
-            //Signaling that the client can perform PKey Auth on supported platforms
+            // Signaling that the client can perform PKey Auth on supported platforms
             if (DeviceAuthHelper.CanOSPerformPKeyAuth())
             {
                 _oAuth2Client.AddHeader(PKeyAuthConstants.DeviceAuthHeaderName, PKeyAuthConstants.DeviceAuthHeaderValue);
             }
 
-            AddExtraHttpHeaders();
+            AddCcsHeader(additionalBodyParameters);
         }
 
-        private void AddExtraHttpHeaders()
+        private void AddCcsHeader(IDictionary<string, string> additionalBodyParameters)
```

Comment:
This was my thought initially when implementing this but  this is not the best way to do this. This is introducing a lot of if statememnts to the token client for flows that do not have CCS headers at all such as the confidential client flows. The token client is perf sensitive code which is why we originally opted to place the CCSHeader creation in the token request classes.

Reply:
can you share the perf baseline? I didn't find it in your PR.

Reply:
The PR was added before the template was added to our PRs where we ask for Perf so I do not have it. We just tried to keep the overall number of conditionals as low as possible.

@bgavrilMS for input

Reply:
but your PR has the template:
![image](https://user-images.githubusercontent.com/19942418/122477600-da485580-cf7c-11eb-9831-07ae068de364.png)
so i wasn't sure what "minor" meant and if you had numbers for that.

Reply:
Oh, that was for the second PR where I was adding the client info to the /authorize endpint. There are 2 PRs for CCS

The original one is where most of the changes occured
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2590

Reply:
Yes, it's less about perf and more about having future-proof code. I thought that each request class should take the responsibility for figuring out the routing hint. 

In the end, I think that only ROPC and IWA are slightly different (because they can use the username), and all the rest of the requests are more or less the same - check if they have an Account object or a LoginHint object. So most of the code was a bunch of if / else anyway https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2713/files#diff-0eecc4a74d3964af4ff79f87b1ca6336318ff7d6bac101461d707c26ab3f27beL272. So I'm not objecting to this anymore, there'll be a ton of if / else either way.

Reply:
Plus for confidential client, the `_requestParams.CcsRoutingHint` will be null, so this whole `if` will be skipped.